### ISSUE TITLE
server

### ISSUE THREAD

**Issue by Specialist22** [koalaman/shellcheck#3201]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2148, SC2215, SC2215, SC2215, SC1036, SC1088
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [x] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
name: Deploy

on:
  push:
    branches:
      - main  # Trigger deployment on push to the main branch
  workflow_dispatch:  # Allow manual triggering

jobs:
  deploy:
    runs-on: ubuntu-latest

    steps:
    - name: Checkout code
      uses: actions/checkout@v3

    - name: Set up Node.js (if deploying a Node app)
      uses: actions/setup-node@v3
      with:
        node-version: '16'  # Specify the Node.js version you need

    - name: Install dependencies
      run: npm install

    - name: Build project
      run: npm run build  # Replace with your build command

    - name: Deploy to Server
      env:
        SERVER: ${{ secrets.SERVER }}  # Set this secret in your repo settings
        USERNAME: ${{ secrets.USERNAME }}  # Set this secret in your repo settings
        SSH_KEY: ${{ secrets.SSH_KEY }}  # Set this secret in your repo settings
      run: |
        ssh -i "${SSH_KEY}" "${USERNAME}"@"${SERVER}" << 'EOF'
        cd /path/to/your/app
        git pull origin main
        npm install
        npm run start
        EOF
```

#### Here's what shellcheck currently says:
```
Line 1	SC2148: Tips depend on target shell and yours is unknown. Add a shebang or a 'shell' directive.
Line 6	SC2215: This flag is used as a command name. Bad line break or missing [ .. ]?
Line 14	SC2215: This flag is used as a command name. Bad line break or missing [ .. ]?
Line 17	SC2215: This flag is used as a command name. Bad line break or missing [ .. ]?
Line 17	SC1036: '(' is invalid here. Did you forget to escape it?
Line 17	SC1088: Parsing stopped here. Invalid use of parentheses?
```
#### Here's what I wanted or expected to see:

TODO: Describe expected/desired output

**Comment by brother** [comment]
Not a shell script.
